FizzBuzz/FizzBuzz/Program.cs

```diff
@@ -27,82 +31,68 @@ public static void print(int number, ArrayList arr)
-                          "Your rules: ");
-            input = Console.ReadLine();
-            string[] rulesString = input.Split(' ');
 
-            foreach (var s in rulesString)
-            {
-                if (s == "3") rule3 = 1;
-                if (s == "5") rule5 = 1;
-                if (s == "7") rule7 = 1;
-                if (s == "11") rule11 = 1;
-                if (s == "13") rule13 = 1;
-                if (s == "17") rule17 = 1;
-            }
-
-            
+        public static void applyRules(int number, List<Rule> rules)
+        {
+            var arr = new ArrayList();
+            bool onlyRule = false;
 
-            for (int number = 1; number <= maxNumber; number++)
+            foreach (Rule rule in rules)
             {
-                
-                var arr = new ArrayList();
-                if (number % 11 == 0 && rule11 == 1)
+                if (number % rule.number == 0)
                 {
-                    if(number % 13 == 0) Console.Write("Fezz");
-                    Console.WriteLine("Bong");
-                    continue;
-                }
-                
-                if (number % 3 == 0 && rule3 == 1) arr.Add("Fizz"); 
-                if (number % 5 == 0 && rule5 == 1) arr.Add("Buzz");
-                if (number % 7 == 0 && rule7 == 1) arr.Add("Bang");
-                
-                if (number % 13 == 0 && rule13 == 1)
-                {
-                    int foundStringStartingWithB = 0;
-                    for (int at = 0; at < arr.Count && foundStringStartingWithB == 0; at++ )
+                    if (rule.onlyThis == true)
                     {
-                        foundStringStartingWithB = insertIfExistsBWord(at, arr);
+                        Console.WriteLine(rule.word);
+                        onlyRule = true;
+                        break;
+                    }
+                    if (rule.priorityBeforeChar != " ")
+                    {
+                        int foundStringSpecifiedChar= 0;
+                        for (int at = 0; at < arr.Count && foundStringSpecifiedChar == 0; at++)
+                        {
+                            foundStringSpecifiedChar = insertIfExistsWord(at, arr,rule.priorityBeforeChar,rule.word);
+                        }
+                        
+                        if (foundStringSpecifiedChar == 0) arr.Add(rule.word);
+                    }
+                    else
+                    {
+                        arr.Add(rule.word);
                     }
-
-                    if (foundStringStartingWithB == 0) arr.Add("Fezz");
                 }
-
-                if (number % 17 == 0 && rule17 == 1) arr.Reverse();
-                
-                print(number,arr);
-
-                
             }
+            if(onlyRule == false) print(number, arr);
         }
+        public static void Main(string[] args)
+        {
+            List<Rule> rules = new List<Rule>();

```

Comment:
a cool way of initializing a list is from an IEnumerable using [this constructor](https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1.-ctor?view=net-6.0#system-collections-generic-list-1-ctor(system-collections-generic-ienumerable((-0)))), e.g.
var rules = new List<Rule>(new [] {
new Rule(3, "Fizz"),
new Rule(5, "Buzz")
...
 })